File: C/1056.cpp
/*author:moonlake23
question1056:组合数的和*/

#include<stdio.h>
int main() {
	int capacity, num1[11],num2[11];
	scanf("%d",&capacity);
	for (int i = 0; i < capacity; i++) {
		scanf("%d", &num1[i]);
		num2[i] = num1[i];
	}

	int all = 0;//总和
	for(int i=0;i<capacity;i++)
		for (int j = 0; j < capacity; j++) {
			if (num1[i] == num2[j])continue;
			else {
				int sum = num1[i] * 10 + num2[j];
				all += sum;
			}
		}
	printf("%d\n", all);
	return 0;
}

File: BasicLevel_Java/1056.java
/*author:moonlake23
 * question1056:组合数的和
 */

import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int cases=sc.nextInt();
		int []num=new int[11];
		for(int i=0;i<cases;i++)
			num[i]=sc.nextInt();
		
		int all=0;
		//num[i]作分母,num[j]作分子
		for(int i=0;i<cases;i++) {
			for(int j=0;j<cases;j++) {
				if(j!=i) {
					int temp=num[i]*10+num[j];
					all+=temp;
				}
			}
		}
		
		System.out.println(all);
		sc.close();
	}

}